compiler/rustc_typeck/src/check/fn_ctxt/_impl.rs

```diff
@@ -1222,19 +1222,21 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         // a problem.
 
         let mut infer_args_for_err = FxHashSet::default();
+
         for &PathSeg(def_id, index) in &path_segs {
             let seg = &segments[index];
             let generics = tcx.generics_of(def_id);
+
             // Argument-position `impl Trait` is treated as a normal generic
             // parameter internally, but we don't allow users to specify the
             // parameter's value explicitly, so we have to do some error-
             // checking here.
             if let GenericArgCountResult {
-                correct: Err(GenericArgCountMismatch { reported: Some(ErrorReported), .. }),
+                correct: Err(GenericArgCountMismatch { reported: Some(_), .. }),
                 ..
-            } = AstConv::check_generic_arg_count_for_call(
-                tcx, span, &generics, &seg, false, // `is_method_call`
-            ) {
+            } =
+                AstConv::check_generic_arg_count_for_call(tcx, span, def_id, &generics, &seg, false)
```

Comment:
Losing the comment explaining that the `false` is is unfortunate, although ideally we should have replaced this with a bespoke enum already :)

Reply:
done 👍🏻